clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2619,9 +2650,19 @@ void CodeGenModule::EmitGlobal(GlobalDecl GD) {
     return;
 
   // If this is an alias definition (which otherwise looks like a declaration)
-  // emit it now.
-  if (Global->hasAttr<AliasAttr>())
-    return EmitAliasDefinition(GD);
+  // handle it now.
+  if (Global->hasAttr<AliasAttr>()) {
+    // Emit the alias here if not for SYCL device.
+    if (!LangOpts.SYCLIsDevice)
+      return EmitAliasDefinition(GD);
+    // Defer for SYCL devices, until either the alias or what it aliases
+    // is used.
+    StringRef MangledName = getMangledName(GD);
+    DeferredDecls[MangledName] = GD;
+    StringRef AliaseeName = Global->getAttr<AliasAttr>()->getAliasee();
```

Comment:
Suggest rewrite `if` on line 2654 and this line with
```
if (AliasAttr *Attr = Global->getAttr<AliasAttr>()) {
  ...
  StringRef AliaseeName = Attr->getAliasee();
```

Reply:
Done.